backend/MonChenil/Services/TimeSlotService.cs

```diff
@@ -57,5 +67,32 @@ private void CheckTimeSlot(TimeSlot timeSlot)
             throw new ArgumentException("Le créneau horaire en chevauche un autre");
         }
     }
-}
 
+    public void AddPetsToTimeSlot(TimeSlot timeSlot, IEnumerable<Pet> pets)
+    {
+        foreach (var pet in pets)
+        {
+            foreach (var existingPet in timeSlot.Pets)
+            {
+                if (!ArePetsCompatible(existingPet, pet))
+                {
+                    throw new ArgumentException($"The pet '{pet.Name}' (Type: {pet.Type}) is not compatible with another pet already booked for this time slot.");
+                }
+            }
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
```

Comment:
Je pense qu'on peut améliorer l'implémentation : 
- mergedPets = pets + timeSlot.Pets // A voir comment on fait en C#
- if (!ArePetsCompatible(mergedPets) { throw } // Ca nécessite de changer la méthode ArePetsCompatible

Faudra mettre à jour GetAvailableTimeSlots si besoin

Reply:
Also did it on the last commit here : #https://github.com/MonChenil/monchenil/commit/0998fd9d611ed4c18275557534b16442494d03a7